actor.go

```diff
@@ -28,89 +25,106 @@ type ActorConfig struct {
 // NewActor returns a new Actor
 func NewActor(c ActorConfig) (*Actor, error) {
 	if c.MailboxSize < 1 {
-		return nil, MailboxSizeTooSmallError("mailbox must be greater than 0")
+		return nil, ActorConfigurationError("mailbox must be greater than 0")
 	}
-	b := &Actor{
-		agent: make(chan Work, c.MailboxSize),
-		quit:  make(chan struct{}),
+	a := &Actor{
+		agent:        make(chan Work, c.MailboxSize),
+		quit:         make(chan struct{}),
+		busy:         new(int32),
+		lastFinished: new(int64),
+		lastAccepted: new(int64),
 	}
-	go b.loop()
-	return b, nil
+	go a.loop()
+	return a, nil
 }
 
 // Accept will push a Letter onto the Actors mailbox. If the box is full, this will
 // block.
-func (b *Actor) Accept(w Work) error {
-	if b.IsShutdown() {
-		return errors.New("Actor is shutting down, no longer accepting work")
+func (a *Actor) Accept(w Work) error {
+	if a.IsShutdown() {
+		return ActorShuttingDownError("actor is shutting down, cannot accept work")
+	}
+	select {
+	case a.agent <- w:
+		atomic.StoreInt64(a.lastAccepted, time.Now().Unix())
+	default:
+		return ActorFullError("actor is full, cannot accept work")
 	}
-	b.agent <- w
-	b.acceptMutex.Lock()
-	b.lastAccepted = time.Now()
-	b.acceptMutex.Unlock()
 	return nil
 }
 
 // LastAccepted returns the last time this Actor got a new letter. Note that while this is
 // protected  by a Mutex, by the time you take action on the result of this value,
 // it may have changed by another concurrent operation.
-func (b *Actor) LastAccepted() time.Time {
-	b.acceptMutex.RLock()
-	defer b.acceptMutex.RUnlock()
-	return b.lastAccepted
+func (a *Actor) LastAccepted() int64 {
+	return atomic.LoadInt64(a.lastAccepted)
 }
 
 // LastFinished returns the last time this Actor completed a job. Note that while this is
 // protected  by a Mutex, by the time you take action on the result of this value,
 // it may have changed by another concurrent operation.
-func (b *Actor) LastFinished() time.Time {
-	b.acceptMutex.RLock()
-	defer b.acceptMutex.RUnlock()
-	return b.lastAccepted
+func (a *Actor) LastFinished() int64 {
+	return atomic.LoadInt64(a.lastFinished)
+
 }
 
 // IsBusy returns if the Actor is currently working or not. Note that while this is
 // protected  by a Mutex, by the time you take action on the result of this value,
 // it may have changed by another concurrent operation.
-func (b *Actor) IsBusy() bool {
-	b.busyMutex.RLock()
-	defer b.busyMutex.RUnlock()
-	return b.busy
+func (a *Actor) IsBusy() bool {
+	if busy := atomic.LoadInt32(a.busy); busy == 1 {
```

Comment:
1 should probably be defined in a const above, and I think this whole clause can be reduced to like `return atomic.LoadInt32(a.busy) == 1` right?